File: core/clientLaya/game/src/shine/dataEx/AffairTimeLock.ts
namespace Shine
{
	/** 事务计时锁(s) */
    export class AffairTimeLock
    {
        private _timeMax:number;

		/** 上次检查时间(s) */
		private _lastCheckTime:number=-1;
		public constructor(time:number=ShineSetting.affairDefaultExecuteTime)
		{
			this._timeMax=time;
			this._lastCheckTime=-1;
		}

		/** 锁定 */
		public lockOn():void
		{
			this._lastCheckTime=DateControl.getTimeSeconds();
		}

		/** 解锁 */
		public unlock():void
		{
			this._lastCheckTime=-1;
		}

		/** 是否锁定中 */
		public isLocking():boolean
		{
			return this._lastCheckTime!=-1 && (DateControl.getTimeSeconds()-this._lastCheckTime)<this._timeMax;
		}
    }
}

File: core/client/game/src/shine/dataEx/AffairTimeLock.cs
namespace ShineEngine
{
	/** 事务计时锁(s) */
	public class AffairTimeLock
	{
		private int _timeMax;

		/** 上次检查时间(s) */
		private long _lastCheckTime=-1L;

		public AffairTimeLock():this(ShineSetting.affairDefaultExecuteTime)
		{

		}

		public AffairTimeLock(int time)
		{
			_timeMax=time;
			_lastCheckTime=-1L;
		}

		/** 锁定 */
		public void lockOn()
		{
			_lastCheckTime=DateControl.getTimeSeconds();
		}

		/** 解锁 */
		public void unlock()
		{
			_lastCheckTime=-1L;
		}

		/** 是否锁定中 */
		public bool isLocking()
		{
			return _lastCheckTime!=-1L && (DateControl.getTimeSeconds()-_lastCheckTime)<_timeMax;
		}
	}
}